qiskit/algorithms/time_evolvers/evolution_problem.py

```diff
@@ -0,0 +1,96 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Evolution problem class."""
+from __future__ import annotations
+from typing import Dict
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import PauliSumOp
+from ..list_or_dict import ListOrDict
+from ...quantum_info.operators.base_operator import BaseOperator
+
+
+class EvolutionProblem:
+    """Evolution problem class.
+
+    This class is the input to time evolution algorithms and must contain information on the total
+    evolution time, a quantum state to be evolved and under which Hamiltonian the state is evolved.
+    """
+
+    def __init__(
+        self,
+        hamiltonian: BaseOperator | PauliSumOp,
+        time: float,
+        initial_state: QuantumCircuit | None = None,
+        aux_operators: ListOrDict[BaseOperator | PauliSumOp] | None = None,
+        truncation_threshold: float = 1e-12,
+        t_param: Parameter | None = None,
+        param_value_dict: Dict[Parameter, complex]
```

Comment:
Would be more flexible for user to typehint/support `Mapping[Parameter, complex]`